packages/client/hmi-client/src/components/model/tera-model.vue

```diff
@@ -102,28 +107,43 @@ useClientEvent(ClientEventType.TaskGollmModelCard, (event: ClientEvent<TaskRespo
 	fetchModel();
 });
 
-const teraModelPartsRef = ref();
-
 const model = ref<Model | null>(null);
+const temporaryModel = ref<Model | null>(null);
+
 const newName = ref('New Model');
 const isRenaming = ref(false);
 const isModelLoading = ref(false);
 const showSaveModal = ref(false);
-
 const isNaming = computed(() => isEmpty(props.assetId) || isRenaming.value);
-const isSaved = computed(() => isEqual(model.value, teraModelPartsRef.value?.transientModel));
+const hasChanged = computed(() => !isEqual(model.value, temporaryModel.value));
+const hasEditPermission = computed(() => useProjects().hasEditPermission());
```

Comment:
maybe this can be a regular variable unless the `hasEditPermission()` updates live